src/field_map_change.c

```diff
@@ -845,34 +826,34 @@ static BOOL sub_02053B44 (TaskManager * taskMan)
 
 static void sub_02053BD4 (TaskManager * taskMan)
 {
-    sub_02050944(taskMan, sub_02053BE4, NULL);
+    FieldTask_Start(taskMan, sub_02053BE4, NULL);
 }
 
 static BOOL sub_02053BE4 (TaskManager * taskMan)
 {
-    int * v0 = sub_02050A68(taskMan);
+    int * state = FieldTask_GetState(taskMan);
 
-    switch (*v0) {
+    switch (*state) {
     case 0:
         sub_02055820(taskMan);
-        (*v0)++;
+        (*state)++;
         break;
     case 1:
-        return 1;
+        return TRUE;
     }
 
-    return 0;
+    return FALSE;
```

Comment:
This looks unreachable to me; is it needed to match?

Reply:
this is absolutely reachable? state = 0 goes to case 0, breaks, then returns FALSE